app/sections/root/sections/tweets/tweets.component.ts

```diff
@@ -6,7 +10,73 @@ import {Component, Input} from 'angular2/core';
   styles: [
     <string> require('./tweets.styl')
   ],
+  directives: [...CORE_DIRECTIVES],
+  pipes: [ToDate]
 })
-export class TweetsComponent {
-  @Input() author:string;
+export class TweetsComponent implements OnInit {
+  @Input()
+  author:string;
+
+  private infFollow: Object= {
+    link: 'https://twitter.com/hansrosling',
+    title: 'Follow Hans',
+    count: 'Followers'
+  };
+
+  private tweets: Array<any> = [];
+  private currentTweet: any;
+  private disabled: boolean = false;
+
+  constructor(private twitter: TwitterService) {
+  }
+
+  ngOnInit():any {
+    this.twitter
+      .create()
+      .author(this.author)
+      .getTweets()
+      .subscribe(this.processTweets.bind(this));
+  }
+
+  nextTweet():any {
+    const foundTweet = _.findLast(this.tweets, (tweet:any) => tweet.id > this.currentTweet.id);
+
+    if (foundTweet) {
+      return this.currentTweet = foundTweet;
+    }
+    this.disabled = true;
+
+    this.twitter
+      .create()
+      .author(this.author)
+      .sinceId(this.currentTweet.id_str)
+      .getTweets()
+      .subscribe(this.processTweets.bind(this));
+  }
+
+  previousTweet():any {
+    const foundTweet = _.find(this.tweets, (tweet:any) => tweet.id < this.currentTweet.id);
+
+    if (foundTweet) {
+      return this.currentTweet = foundTweet;
+    }
+    this.disabled = true;
+
+    this.twitter
+      .create()
```

Comment:
could we wrap this and https://github.com/Gapminder/gapminder-org/pull/35/files#diff-d35d465ebf2de003ea9e2e1a968f7a33R50 and https://github.com/Gapminder/gapminder-org/pull/35/files#diff-d35d465ebf2de003ea9e2e1a968f7a33R35 into separated function ? ( about 60% is the same )